### PR TITLE
Upgrade SvelteKit to 1.0

### PR DESCRIPTION
**Pull request by AndrewLester**
## Status:

🚀 : Ready

## Description

- Upgrades SvelteKit to version 1.0
- Crazy that this ~netted 1 line~ reduced line count 😯

Things to think about:
- I removed the `loadImages` script from the build process, and everything still seems to work. I was wondering what the reason for it was before, if it was for an old SvelteKit bug or something to do with the Contentful image URLs? The reason I removed it is because I'd have to change the file paths it accesses since they're no longer valid. I _could_ do this though.
- I switched to the SvelteKit Netlify adapter because I'm anticipating some need for server-side JS soon (viewing preview content). All of the old content on the site is still prerendered, but now the build will include server-side JS for use in the future.
- For some reason the Netlify adapter is still able to deploy to Vercel... I have no idea why. Not a bad thing, just a thing.
- Had to disable comment removal minification option because Svelte has [an issue](https://github.com/sveltejs/kit/issues/7493#issuecomment-1303218382) with it.
- I was able to simplify the use of the `<Head` component by having pages return load data called `title`, and the layout uses it for setting a global `<Head`. While this adds some indirection, I think it's fine because of how much it simplifies, and TypeScript + autocomplete makes `title` appear inside the return statements of `load` functions.
- I had to pass the Contentful private environment keys into the `check` CI/CD jobs because they're statically replaced variables, so they have to be there for SvelteKit to successfully type check our app.

### CODE REVIEW

**Comment by arpanlaha** [comment]
@AndrewLester `loadImages` is for backwards compatibility with our old website which hosted member images, as a lot of our GitHub projects used our website's pictures in their readmes - so ideally we'd maintain the functionality. 

Here's an example URL: `https://uiuc.hack4impact.org/images/people/arpan_laha.jpg`.

**Comment by arpanlaha** [comment]
Why are the file paths no longer valid, and how would you change them? TBH whatever works as long as the output is high-resolution images matching the directory/file structure that we have today so that we have easily-accessible content without resorting to Contentful's API.

**Comment by AndrewLester** [comment]
> Why are the file paths no longer valid, and how would you change them?

EDIT: By file paths here, I mean the file paths that the script _reads_ from, the written ones are still fine.

The file paths are no longer valid since they used to be from "data endpoints" with URLs that end in `.json` like `build/server/members.json`. Now, the way SvelteKit is being used here includes page endpoints with URLs that end in `/__data.json`. I'd just need to change these file paths.

Now that I think about it though, since we're shipping server-side JS, I could just set up an API endpoint to route these image requests through to Contentful and set a huge cache control max age or something. That way no script will have to read built file contents (though since SvelteKit is now stable these shouldn't change anyways).

**Comment by arpanlaha** [comment]
> Now that I think about it though, since we're shipping server-side JS, I could just set up an API endpoint to route these image requests through to Contentful and set a huge cache control max age or something. That way no script will have to read built file contents (though since SvelteKit is now stable these shouldn't change anyways).

I would say to optimize for a balance of simplicity/performance here - if we're able to get things working by just swapping the input file paths that seems to me to be best here since it reduces the work done on-demand. I'll defer to your judgement though.

**Comment by AndrewLester** [comment]
I opted to create image endpoints since I checked out the JSON data we now have from the build, and its structure is really hard to understand. Seems to be a complex internal SvelteKit API. The image endpoints turned out pretty well, they even support height query arguments now. They also forward the caching headers straight from Contentful, the caching is good to go.

Side note, you can't use the Vercel preview deployment to see the image endpoints since the Netlify adapter is currently being used. Have to use the Netlify preview.

File: package.json [hunk 1]
```diff
@@ -3,47 +3,42 @@
...
-    "@sveltejs/adapter-static": "1.0.0-next.37",
-    "@sveltejs/kit": "1.0.0-next.379",
+    "@sveltejs/adapter-netlify": "^1.0.0",
+    "@sveltejs/kit": "^1.0.0",
     "@types/ackee-tracker": "^5.0.1",
     "@types/html-minifier-terser": "^6.1.0",
     "@typescript-eslint/eslint-plugin": "^5.30.6",
     "@typescript-eslint/parser": "^5.30.6",
-    "dotenv": "^10.0.0",
+    "contentful": "^9.1.5",
```
Comment (arpanlaha) [inline package.json]: nit: doesn't really matter but I feel like this belongs in `dependencies` more than `devDependencies`

File: src/lib/components/Button.svelte [hunk 2]
```diff
@@ -34,15 +34,10 @@
...
-
-  $: isExternal =
-    href?.startsWith("http") ||
-    href?.startsWith("mailto") ||
-    href?.startsWith("tel");
 </script>
 
 {#if href}
-  <a {...props} {href} sveltekit:prefetch={!isExternal ? true : undefined}>
+  <a {...props} {href}>
```
Comment (arpanlaha) [inline src/lib/components/Button.svelte:40]: why remove prefetch? is this something done automatically now?

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -1,57 +1,22 @@
...
-  url="https://uiuc.hack4impact.org"
-  image={partnerships.src}
-/>
-
 <DoubleBanner leftColor="var(--blue)" rightColor="var(--blue-darker)">
   <span slot="left">
-    {#if banner.text && banner.url}
-      <a class="banner" href={banner.url} sveltekit:prefetch>
-        {banner.text}
+    {#if data.info.homepageBannerText && data.info.homepageBannerUrl}
```
Comment (arpanlaha) [inline src/routes/+page.svelte:17]: Does this have type info? More a question about the framework than your code

**Review by arpanlaha — COMMENTED** [review]
Looks good to me overall, but want to make sure I understand a few changes in the framework

File: src/lib/components/Button.svelte (hunk 2 above)
Comment (AndrewLester) [reply src/lib/components/Button.svelte:40]: Yep, it's set in the `body` tag here:

https://github.com/hack4impact-uiuc/website/blob/893f20050b317af4bf2089baf3d18b2f01eb438e/src/app.html#L265

This then applies prefetching to all `a` elements in the subtree.

File: src/routes/+page.svelte (hunk 3 above)
Comment (AndrewLester) [reply src/routes/+page.svelte:17]: Yep, all the type info in `data` is generated from the return types of load functions of the current page + parent layouts.

**Review by arpanlaha — APPROVED** [review]
sweet, thanks for the hard work here!

File: package.json (hunk 1 above)
Comment (AndrewLester) [reply package.json]: I'll move this back, I thought SvelteKit might be doing something weird with prod vs dev dependencies, but everything still works.

**Pull request merged**